tests/common.go

```diff
@@ -7213,10 +7213,36 @@ func EndPxBackupTorpedoTest(contexts []*scheduler.Context) {
 	if TestRailSetupSuccessful && CurrentTestRailTestCaseId != 0 && RunIdForSuite != 0 {
 		AfterEachTest(contexts, CurrentTestRailTestCaseId, RunIdForSuite)
 	}
+
 	ginkgoTestDescr := ginkgo.CurrentGinkgoTestDescription()
 	if ginkgoTestDescr.Failed {
 		log.Infof(">>>> FAILED TEST: %s", ginkgoTestDescr.FullTestText)
 	}
+
+	// Cleanup all the namespaces created by the testcase
+	err := DeleteAllNamespacesCreatedByTestCase(CurrentTestRailTestCaseId)
```

Comment:
@kshithijiyer-px Let me know your thoughts. Should we delete the namespace only if the test case has passed? If the test case fails we would want the restored apps to still be present for debugging.

Reply:
Say if 10 cases fail and we have hit the max volume per node this will cause the next 40 to fail as well.

Reply:
Makes sense

Reply:
But can we do this only for IBM clusters or any cluster which has this restriction. For these reasons:

- We can still have the namespaces and application running after the run in case we need to debug anything on our on prem clusters
- We can save a lot of time in our on prem pipelines since we will be avoiding cluster switching

Reply:
This will be the case everywhere as l lf them have some sort of limits after which this will happen yes the number can be higher but sooner or later this will happen.

Reply:
Yes, but we are already consuming too much time in pipelines. So at least we can avoid this switching in on prem clusters. We will make this generic once we start hitting this issue on on-prem clusters

Reply:
I feel it's better to do it now itself then get into that state and then enable the pr again and test the same again.

Reply:
It will happen on on-premise as well if we enable ocs as the max volumes per node will be the same as what we see in IBM. My only intention of doing it everywhere not itself is to save sometime which we would have spent in debugging the issue again on on-premise once we reach that volume.

Reply:
Then let us do it the other way round. Lets us not run this only on Vanilla/RKE/OCP. Everywhere else we can delete the namespace.

Are we sure that OCS has the limit? What is that number?

Reply:
I know this is not for OCS but we you can check this https://kubernetes.io/docs/concepts/storage/storage-limits/#kubernetes-default-limits

Reply:
Based on offline discussion with @mkoppal-px and results which we saw while running this jobs with and without this PR, we are unable to conclude if this will help or harm the runs by increasing the execution time due to the switch which happens for moving between destination and source cluster. 

For which we have decided to go with the PR as is and if it's causing issue more time to run we'll update the code to skip in case of on-premise.